ako-gateway-api/nodes/gateway_model_rel.go

```diff
@@ -258,7 +264,21 @@ func HTTPRouteToGateway(namespace, name, key string) ([]string, bool) {
 	var gatewayList []string
 	var gwNsNameList []string
 	parentNameToHostnameMap := make(map[string][]string)
+	statusIndex := 0
+	httpRouteStatus := akogatewayapiobjects.GatewayApiLister().GetRouteToRouteStatusMapping(routeTypeNsName)
 	for _, parentRef := range hrObj.Spec.ParentRefs {
+		if statusIndex >= len(httpRouteStatus.Parents) {
```

Comment:
Is following scenario possible?
httpRouteStatus.Parents array --> parent1, parent2
hrObj.Spec.ParentRefs array --> parent2, parent1?

If such scenario is possible, should we reset status index to 0?

Reply:
It will never happen, since whenever there will be an update on HTTPRoute, its revalidation will go through and in revalidation we go index by index and then update httpRouteStatus